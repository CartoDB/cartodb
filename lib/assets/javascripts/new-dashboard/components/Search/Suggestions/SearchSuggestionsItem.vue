<template>
  <a
    :href="visualizationURL"
    class="suggestions__item text is-caption"
    :class="`suggestions__item--${item.type}`"
    @click="onItemClicked">
    {{item.name}}
  </a>
</template>

<script>
import * as Visualization from 'new-dashboard/core/visualization';

export default {
  name: 'SearchSuggestionsItem',
  props: {
    item: Object
  },
  computed: {
    visualizationURL () {
      return Visualization.getURL(this.$props.item, this.$cartoModels);
    }
  },
  methods: {
    onItemClicked () {
      this.$emit('itemClick');
    }
  }
};
</script>

<style lang="scss" scoped>
@import 'stylesheets/new-dashboard/variables';

.suggestions__item {
  display: block;
  position: relative;
  width: 100%;
  padding: 16px 16px 16px 36px;
  overflow: hidden;
  color: $text-color;
  text-overflow: ellipsis;
  white-space: nowrap;
  cursor: pointer;

  &:hover {
    background-color: rgba($primary-color, 0.05);
    color: #1785FB;
    text-decoration: none;
  }

  &::after {
    content: '';
    position: absolute;
    top: 50%;
    left: 14px;
    width: 12px;
    height: 100%;
    transform: translate3d(0, -50%, 0);
    background-repeat: no-repeat;
    background-position: center left;
    background-size: contain;
  }
}

.suggestions__item--derived {
  &::after {
    background-image: url('../../../assets/icons/navbar/search/search-map.svg');
  }
}

.suggestions__item--table {
  &::after {
    background-image: url('../../../assets/icons/navbar/search/search-data.svg');
  }
}
</style>
