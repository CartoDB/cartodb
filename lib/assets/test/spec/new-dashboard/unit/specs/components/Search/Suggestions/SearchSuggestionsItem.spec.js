import { shallowMount } from '@vue/test-utils';
import SearchSuggestionsItem from 'new-dashboard/components/Search/Suggestions/SearchSuggestionsItem';
import visualizationData from '../../../fixtures/visualizations';
import usersArray from '../../../fixtures/users';

import UserModel from 'dashboard/data/user-model';
import ConfigModel from 'dashboard/data/config-model';

describe('SearchSuggestionsItem.vue', () => {
  let $cartoModels, suggestionItem;

  beforeEach(() => {
    $cartoModels = configCartoModels({ user: usersArray[1] });

    suggestionItem = shallowMount(SearchSuggestionsItem, {
      propsData: { item: visualizationData.visualizations[0] },
      mocks: { $cartoModels }
    });
  });

  it('should render correctly', () => {
    expect(suggestionItem).toMatchSnapshot();
  });

  it('should emit itemClick event when suggestion is clicked', () => {
    suggestionItem.find('.suggestions__item').trigger('click');

    expect(suggestionItem.emitted('itemClick')).toBeTruthy();
  });
});

function configCartoModels (attributes = {}) {
  const user = new UserModel(attributes.user || usersArray[0]);
  const config = new ConfigModel(attributes.config || { maps_api_template: 'http://{user}.example.com' });
  return { user, config };
}
