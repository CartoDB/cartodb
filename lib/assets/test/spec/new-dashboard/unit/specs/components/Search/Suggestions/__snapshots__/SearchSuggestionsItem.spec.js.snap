// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSuggestionsItem.vue should render correctly 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200/map" class="suggestions__item text is-caption suggestions__item--derived">
  First Map
</a>
`;
